Android NDK support improvement (#327)
cloudflare/quiche: README.md

To build quiche for Android, you need the following:

> - Install [Android NDK] (13b or higher), using Android Studio or directly.

Reviewer: ```suggestion
- Install the [Android NDK] (13b or higher), using Android Studio or directly.
```

- Set `ANDROID_NDK_HOME` environment variable to NDK path, e.g.

```bash
 $ export ANDROID_NDK_HOME=/usr/local/share/android-ndk
```

- Install the Rust toolchain for Android architectures needed:

```bash
 $ rustup target add aarch64-linux-android arm-linux-androideabi armv7-linux-androideabi i686-linux-android x86_64-linux-android
```

Note that the minimum API level is 21 for all target architectures.

> Depending on NDK version, you can take one of the following way.

Reviewer: ```suggestion
Depending on the NDK version used, you can take one of the following procedures:
```

[Android NDK]: https://developer.android.com/ndk

#### NDK version >= 19

For NDK version 19 or higher (21 recommended), you can build in a simpler
way using [cargo-ndk]. You need to install [cargo-ndk] first.

```bash
 $ cargo install cargo-ndk
```

> You can build quiche library using the following way. Note that

Reviewer: ```suggestion
You can build the quiche library using the following procedure. Note that
```

`--target` and `--android-platform` are mandatory.

```bash
 $ cargo ndk --target aarch64-linux-android --android-platform 21 -- build
```

> See [build_android_ndk19.sh] for more example.

Reviewer: ```suggestion
See [build_android_ndk19.sh] for more information.
```

> Note that building with NDK version 18 seems broken.

Reviewer: ```suggestion
Note that building with NDK version 18 appears to be broken.
```
